mbtiles/scripts/cli.py

```diff
@@ -348,10 +397,29 @@ def mbtiles(
             )
             ratio = min_ratio = raster_area / minzoom_tile_area
 
-            while ratio < 16:
-                est_num_tiles += len(
-                    list(mercantile.tiles(west, south, east, north, [zoom]))
+            # If given a cutline, we use its mercator area and the
+            # supermercado module to help estimate the number of output
+            # tiles.
+            if cutline:
+                geoms = [shape(f["geometry"]) for f in cutline]
+                union = unary_union(geoms)
+                cutline_mercator = transform_geom(
+                    "OGC:CRS84", "EPSG:3857", mapping(union)
                 )
+                min_ratio *= shape(cutline_mercator).area / raster_area
+                ratio = min_ratio
+                estimator = functools.partial(supermercado.burntiles.burn, cutline)
```

Comment:
Here is the third aspect of the cutline: we use it to make an estimate of the number of output tiles.